index.d.ts

```diff
@@ -0,0 +1,51 @@
+export interface ActionCreator {
+  (...args: any[]): any;
+}
+
+export type Reducer<S> = (state: S, action: any) => S;
+
+export type Dispatch = (action: any) => any;
+
+export interface MiddlewareAPI<S> {
+  dispatch: Dispatch;
+  getState: () => S;
+}
+
+export interface Middleware {
+  <S>(api: MiddlewareAPI<S>): (next: Dispatch) => Dispatch;
+}
+
+export class Store<S> {
```

Comment:
It's not a class because you can not import it from Redux (or 'redux') lib…

Trying this:

```
import { Store } from "redux";
const myStore = new Store({myStorKeys:andValues});
```

or just

```
const myStore = new Redux.Store({myStorKeys:andValues});
```

will compile but throw an error in runtime - this is makes static typing a bit useless :(